src/georaster-layer-for-leaflet.ts

```diff
@@ -321,12 +334,30 @@ const GeoRasterLayer: (new (options: GeoRasterLayerOptions) => any) & typeof L.C
     // note that we aren't setting the tile height or width here
     // drawTile dynamically sets the width and padding based on
     // how much the georaster takes up the tile area
-    this.drawTile({ tile, coords, context, done });
+    const coordsKey = this._tileCoordsToKey(coords);
+
+    const resolution = this._getResolution(coords.z);
+    const key = `${coordsKey}:${resolution}`;
+    const doneCb = (error?: Error, tile?: HTMLElement): void => {
+      done(error, tile);
+
+      // caching the rendered tile, to skip the calculation for the next time
+      if (!error && this.options.caching) {
+        this.cache[key] = tile;
+      }
+    };
+
+    if (this.options.caching && this.cache[key]) {
+      done(undefined, this.cache[key]);
```

Comment:
Calling done() directly violates the expectation of GridLayer (see https://github.com/Leaflet/Leaflet/blob/93f681c488ec07690cd4a393056078cf88f3bcff/src/layer/tile/GridLayer.js#L816C2-L818C2) and in my tests causes the old tiles to disappear before the new ones are drawn. I'd suggest emulating the behavior of drawTile() with `setTimeout(async () => done(undefined, this.cache[key]), 0);`